cmd/gateway-gcs.go

```diff
@@ -90,6 +114,12 @@ func gcsToObjectError(err error, params ...string) error {
 	if !ok {
 		// We don't interpret non Minio errors. As minio errors will
 		// have StatusCode to help to convert to object errors.
+		e.e = err
+		return e
+	}
+
+	if len(googleAPIErr.Errors) == 0 {
+		e.e = err
```

Comment:
Looks like these changes took care of the panic, however, error messages like the following are printed (when doing the same test):

```
ERRO[1098] Unable to create object part.                 cause=googleapi: got HTTP response code 400 with body: Client sent query request with a non-empty body. source=[object-handlers.go:817:objectAPIHandlers.PutObjectPartHandler()] stack=home/aditya/Code/goenv-1.7.5/workspace/src/github.com/minio/minio/cmd/gateway-gcs.go:661:(*gcsGateway).PutObjectPart home/aditya/Code/goenv-1.7.5/workspace/src/github.com/minio/minio/cmd/object-handlers.go:814:objectAPIHandlers.PutObjectPartHandler home/aditya/Code/goenv-1.7.5/workspace/src/github.com/minio/minio/cmd/api-router.go:46:(objectAPIHandlers).PutObjectPartHandler-fm
```

The API response seems to suggest that something like a `GET` request was done with a non-empty body - which is weird. The upload completes fine, because I see that the part that got the above error response is simply retried by `mc`. This seems to be a bug of some kind - either in our code or in the google storage api golang library.